common/include/pcl/point_types_conversion.h

```diff
@@ -136,12 +136,13 @@ namespace pcl
     if (out.h < 0.f) out.h += 360.f;
   }
 
-  /** \brief Convert a XYZRGBA point type to a XYZLAB
-    * \param[in] in the input XYZRGBA point
+  /** \brief Convert a XYZRGB-based point type to a XYZLAB
+    * \param[in] in the input XYZRGB(XYZRGBA, XYZRGBL, etc.) point
     * \param[out] out the output XYZLAB point
     */
+  template <typename PointXYZRGBTypes>
```

Comment:
Please constrain for eg using `HasColor<PointT> = true`

PS: @mvieth thoughts on changing the name to `has_rgb`?

Reply:
Thanks for your advice, I'd add `static_assert` in the function as a constraint.:smiley: And if `has_color` needs a rename, I will commit new changes.

Reply:
```suggestion
  template <typename PointT, HasColor<PointT> = true>
```
No, not an assert. That'd make it impossible to overload in future for some-random reason.

Reply:
It's really a better way, thanks for your kind help!

Reply:
> PS: @mvieth thoughts on changing the name to `has_rgb` from `has_color` in point_types.hpp?

Seems like a good idea. `has_rgb` should check for rgb and rgba, while `has_color` could check for any color representation, e.g. rgb, rgba, hsv, lab. But can we just change that without breaking anything? Is it maybe necessary to add `has_rgb` and deprecate `has_color` (and optionally add a new struct, e.g. `has_any_color`)?

Reply:
Thanks for your reviewing!

I've investigated the reference of `has_color` and related `has_color_v` `HasColor` `HasNoColor` in pcl, and found it occurs 23 times in 7 files (includes 2 test codes). It seems not very difficult for me to change them all, if there isn't some other reasons(not very sure).
<details>
<summary>Detailed search result</summary>

```text
23 results - 7 files

common/include/pcl/common/impl/accumulators.hpp:
  139:       using IsCompatible = pcl::traits::has_color<boost::mpl::_1>;

common/include/pcl/common/impl/copy_point.hpp:
  92:                                                          boost::mpl::or_<boost::mpl::not_<pcl::traits::has_color<PointInT>>,
  93:                                                                          boost::mpl::not_<pcl::traits::has_color<PointOutT>>,

common/include/pcl/impl/point_types.hpp:
  2354:   struct has_color : has_any_field<PointT, boost::mpl::vector<pcl::fields::rgb,
  2359:   constexpr auto has_color_v = has_color<PointT>::value;
  2362:   using HasColor = std::enable_if_t<has_color_v<PointT>, bool>;
  2365:   using HasNoColor = std::enable_if_t<!has_color_v<PointT>, bool>;

test/common/test_common.cpp:
  411:   // has_color
  412:   EXPECT_TRUE ((pcl::traits::has_color<pcl::PointXYZRGB>::value));
  413:   EXPECT_TRUE ((pcl::traits::has_color<pcl::PointXYZRGBA>::value));
  414:   EXPECT_FALSE ((pcl::traits::has_color<pcl::PointXYZ>::value));

test/common/test_type_traits.cpp:
  114: TEST (TypeTraits, HasColor)
  116:   static_assert(!pcl::traits::has_color_v<pcl::PointXYZ>,
  117:                 "has_color<> should detect lack of rgb field");
  118:   static_assert(pcl::traits::has_color_v<pcl::PointXYZRGB>,
  119:                 "has_color<> should detect rgb field");
  120:   static_assert(pcl::traits::has_color_v<pcl::PointXYZRGBA>,
  121:                 "has_color<> should detect rgb field");

tools/real_sense_viewer.cpp:
  316:       if (pcl::traits::has_color<PointT>::value)

visualization/include/pcl/visualization/impl/pcl_visualizer.hpp:
  99:   if (pcl::traits::has_color<PointT>())

```

</details>

And could I just change current `has_color` to `has_rgb` and add a new struct `has_color` to check for any color representation? If so, I'm pleased to work on it in another PR! :blush:

Reply:
Let's move the change of color/rgb trait to separate PR.

I agree with @mvieth that color should be deprecated, instead of a behavior change, and `has_any_color` brought in as the new trait.

Reply:
Thank you! I'd work on that in another PR.